### DISCUSSION TITLE
Cant get simple playlist to work on 2.1.4

### DISCUSSION THREAD

**Discussion by StudentSA** [savonet/liquidsoap#2991] [Q&A]
I wanted to upgrade from 1.4.4 to 2.1.4 however I was unable to run a script that worked on 1.4.4

the line:

`local_files=playlist("/tmp/playlist.pls");`

fails with error:

    At activestream.liq, line 3, char 12-21:
    local_files=playlist("/tmp/playlist.pls");

    Error 5: this value has type
      unit
    but it should be a subtype of
      (_) -> _

What am i missing?

**Comment by vitoyucepi** [comment]
Hi @StudentSA,

Probably the error is not in the script but in the installation.
Could you describe the way you're installing liquidsoap?

Have you tested your liquidsoap script in the docker container?

There's also a migration guide [here](https://www.liquidsoap.info/doc-2.1.4/migrating.html).

**Comment by StudentSA** [comment]
Have not tested docker.

I used a fresh 22.04 Ubuntu cloud instance.

    apt update and apt upgrade
    apt intsall opam

    opam init
    opam update
    opam install opus taglib mad lame cry ffmpeg liquidsoap

    liquidsoap --version
    Liquidsoap 2.1.4
    Copyright (c) 2003-2022 Savonet team
    Liquidsoap is open-source software, released under GNU General Public License.
    See <http://liquidsoap.info> for more information.

**Comment by vitoyucepi** [comment]
Looks fine to me, but I can't replicate this behavior locally.

Could you check if this script works on your end?
```ruby
local_files = playlist("/tmp/playlist.pls")
output.dummy(fallible=true, local_files)
```

**Comment by StudentSA** [comment]
Okay, so I found the issue.

I needed to use liquidsoap from another user on linux, so I copied the binary to the global bin folder.

    sudo cp ~/.opam/default/bin/liquidsoap /usr/bin/liquidsoap

turns out this might not be the correct way, when reverting this change liquidsoap works as expected.

How does one use opam to install liquidsoap for all users on a linux machine?

**Reply by vitoyucepi** [reply]
Your question is not a simple one.
As far as I understand, opam doesn't support system-wide installation. 
ocaml/opam#1950.

But you can install the liquidsoap deb package for *ubuntu:22.04*.
You can get it among other release artifacts [here](https://github.com/savonet/liquidsoap/releases/tag/v2.1.4).

Also, it's possible to use liquidsoap in a docker container.
I think you can configure it to run from any user.

Finally, you can allow other users to execute the liquidsoap binary.
Maybe symlinking your binary into `/usr/bin` will do the trick.
